### PR TITLE
update block test #2478

### PR DESCRIPTION
**Pull request by millken**
#2478

### CODE REVIEW

File: blockchain/block/blockstore_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	store, err := makeStore()
+	require.NoError(err)
+
+	storeProto := store.ToProto()
+
+	require.NotNil(storeProto)
+	require.Equal(0, len(storeProto.Receipts))
+
+	err = store.FromProto(storeProto)
+	require.NoError(err)
```
Comment (CoderZhi) [inline blockchain/block/blockstore_test.go:30]: require.NoError(store.FromProto(storeProto))

File: blockchain/block/blockstore_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func TestStoreSerDer(t *testing.T) {
+	require := require.New(t)
+	store, err := makeStore()
+	require.NoError(err)
+
+	ser, err := store.Serialize()
+	require.NoError(err)
+	require.NoError(store.Deserialize(ser))
```
Comment (CoderZhi) [inline blockchain/block/blockstore_test.go:40]: compare the deserialized store with the original store.

File: blockchain/block/validator_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+func TestValidator(t *testing.T) {
+	require := require.New(t)
+
+	ctx := context.Background()
+	blk := &Block{}
+
+	v := NewValidator(nil)
+	err := v.Validate(ctx, blk)
+	require.NoError(err)
```
Comment (CoderZhi) [inline blockchain/block/validator_test.go:24]: require.NoError(v.Validate(ctx, blk))

Comment (CoderZhi) [reply blockchain/block/validator_test.go:24]: add cases which return error

File: blockchain/block/blockstore_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+}
+
+func makeStore() (store Store, err error) {
+	nblk, err := NewTestingBuilder().
+		SetHeight(1).
+		SetPrevBlockHash(hash.ZeroHash256).
```
Comment (dustinxie) [inline blockchain/block/blockstore_test.go:47]: add SetReceipts()

File: blockchain/block/blockstore_test.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		SetTimeStamp(testutil.TimestampNow()).
+		SignAndBuild(identityset.PrivateKey(29))
+	if err != nil {
+		return
+	}
+
+	store = Store{
+		Block: &nblk,
```
Comment (dustinxie) [inline blockchain/block/blockstore_test.go:55]: add 2 receipts

File: blockchain/block/blockstore_test.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	return
```
Comment (dustinxie) [inline blockchain/block/blockstore_test.go:57]: return store, nil

File: blockchain/block/blockstore_test.go [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	store, err := makeStore()
+	require.NoError(err)
+
+	ser, err := store.Serialize()
+	require.NoError(err)
+	require.NoError(store.Deserialize(ser))
+
+}
+
+func makeStore() (store Store, err error) {
```
Comment (dustinxie) [inline blockchain/block/blockstore_test.go:44]: suggest remove named return, not used often in codebase

**Pull request closed without merging**